src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -60,104 +62,93 @@ private MsQuicApi(QUIC_API_TABLE* apiTable)
                     }
 
-                    Tls13ServerMayBeDisabled = IsTls13Disabled(isServer: true);
-                    Tls13ClientMayBeDisabled = IsTls13Disabled(isServer: false);
+                    return;
                 }
 
-                IsQuicSupported = true;
-            }
-            finally
-            {
-                // Gracefully close the API table to free resources. The API table will be allocated lazily again if needed
-                bool closed = TryCloseMsQuic(msQuicHandle, apiTable);
-                Debug.Assert(closed, "Failed to close MsQuic");
+                Tls13ServerMayBeDisabled = IsTls13Disabled(isServer: true);
+                Tls13ClientMayBeDisabled = IsTls13Disabled(isServer: false);
             }
+
+            IsQuicSupported = true;
         }
         finally
         {
-            // Unload the library, we will load it again when we actually use QUIC
-            NativeLibrary.Free(msQuicHandle);
+            // Gracefully close the API table to free resources. The API table will be allocated lazily again if needed
+            bool closed = TryCloseMsQuic(apiTable);
+            Debug.Assert(closed, "Failed to close MsQuic");
         }
     }
 
     private static MsQuicApi AllocateMsQuicApi()
     {
         Debug.Assert(IsQuicSupported);
 
-        int openStatus = MsQuic.QUIC_STATUS_INTERNAL_ERROR;
-
-        if (TryLoadMsQuic(out IntPtr msQuicHandle) &&
-            TryOpenMsQuic(msQuicHandle, out QUIC_API_TABLE* apiTable, out openStatus))
+        if (!TryOpenMsQuic(out QUIC_API_TABLE* apiTable, out int openStatus))
         {
-            return new MsQuicApi(apiTable);
+            throw ThrowHelper.GetExceptionForMsQuicStatus(openStatus);
         }
 
-        ThrowHelper.ThrowIfMsQuicError(openStatus);
-
-        // this should unreachable as TryOpenMsQuic returns non-success status on failure
-        throw new Exception("Failed to create MsQuicApi instance");
+        return new MsQuicApi(apiTable);
     }
 
     private static bool TryLoadMsQuic(out IntPtr msQuicHandle) =>
         NativeLibrary.TryLoad($"{Interop.Libraries.MsQuic}.{MsQuicVersion.Major}", typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out msQuicHandle) ||
         NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out msQuicHandle);
 
-    private static bool TryOpenMsQuic(IntPtr msQuicHandle, out QUIC_API_TABLE* apiTable, out int openStatus)
+    private static bool TryOpenMsQuic(out QUIC_API_TABLE* apiTable, out int openStatus)
     {
+        Debug.Assert(MsQuicHandle != IntPtr.Zero);
+
         apiTable = null;
-        if (!NativeLibrary.TryGetExport(msQuicHandle, "MsQuicOpenVersion", out IntPtr msQuicOpenVersionAddress))
+        if (!NativeLibrary.TryGetExport(MsQuicHandle, "MsQuicOpenVersion", out IntPtr msQuicOpenVersionAddress))
```

Comment:
I think this should be `NativeLibrary.GetExport`.

The failures to get exports from the MsQuic library should be fatal. If we fail to get the export, it means that there is something very wrong.